app/events/domains/proxy_domains_changed_event.rb

```diff
@@ -0,0 +1,29 @@
+require 'uri'
+
+class Domains::ProxyDomainsChangedEvent < BaseEventStoreEvent
+  def self.create(proxy)
+    new(
+      proxy:    proxy,
+      staging_domains: extract_domain(proxy.sandbox_endpoint),
+      production_domains: extract_domain(proxy.endpoint),
+
+      metadata: {
+        provider_id: (provider_id = proxy.provider&.id),
+        zync: {
+          tenant_id: proxy.tenant_id || proxy.provider&.tenant_id || provider_id,
+          service_id: proxy.service_id,
+        }
+      }
+    )
+  end
+
+  def self.valid?(proxy)
+    !!proxy # TODO: check if deployment_option or domains actually changed
```

Comment:
Imo this method shouldn't be necessary right now. If `proxy` is `nil` or `false`, which I think it is impossible with the current code, it would fail in the create method as soon as it does the `proxy.sandbox_endpoint` 😄

Reply:
I think it won't even start creating the event until this passes. So this would prevent it from crashing. And after ER1 we could try to see how to optimize this and filter only events that matter (like that domain or deployment option changed).